internal/app/maintenance/server/server.go

```diff
@@ -66,3 +68,15 @@ func (s *Server) ApplyConfiguration(ctx context.Context, in *machine.ApplyConfig
 
 	return reply, nil
 }
+
+// GenerateConfiguration implements the machine.MachineServer interface.
+// nolint:gocyclo
+func (s *Server) GenerateConfiguration(ctx context.Context, in *machine.GenerateConfigurationRequest) (reply *machine.GenerateConfigurationResponse, err error) {
+	machineType := v1alpha1machine.Type(in.MachineType)
+
+	if machineType != v1alpha1machine.TypeInit {
```

Comment:
this is probably not true, `TypeJoin` can't be generated, but it's perfectly fine to use `TypeControlPlane` and bootstrap via the API